dipy/align/scalespace.py

```diff
@@ -0,0 +1,422 @@
+
+    def get_scaling(self, level):
+        r"""Adjustment factor for input-spacing to reflect voxel sizes at level
+
+        Returns the scaling factor that needs to be applied to the input
+        spacing (the voxel sizes of the image at level 0 of the scale space) to
+        transform them to voxel sizes at the requested level.
+
+        Parameters
+        ----------
+        level : int, 0 <= from_level < L, (L = number of resolutions)
+            the scale space level to get the scalings from
+
+        Returns
+        -------
+        the scaling factors from the original spacing to the spacings at the
+        requested level
+
+        """
+        return self._get_attribute(self.scalings, level)
+
+    def get_affine(self, level):
+        r"""Voxel-to-space transformation at a given level
+
+        Returns the voxel-to-space transformation associated to the sub-sampled
+        image at a particular resolution of the scale space (note that this
+        object does not explicitly subsample the smoothed images, but only
+        provides the properties the sub-sampled images must have).
+
+        Parameters
+        ----------
+        level : int, 0 <= from_level < L, (L = number of resolutions)
+            the scale space level to get affine transform from
+
+        Returns
+        -------
+            the affine (voxel-to-space) transform at the requested resolution
+            or None if an invalid level was requested
+        """
+        return self._get_attribute(self.affines, level)
+
+    def get_affine_inv(self, level):
+        r"""Space-to-voxel transformation at a given level
+
+        Returns the space-to-voxel transformation associated to the sub-sampled
+        image at a particular resolution of the scale space (note that this
+        object does not explicitly subsample the smoothed images, but only
+        provides the properties the sub-sampled images must have).
+
+        Parameters
+        ----------
+        level : int, 0 <= from_level < L, (L = number of resolutions)
+            the scale space level to get the inverse transform from
+
+        Returns
+        -------
+        the inverse (space-to-voxel) transform at the requested resolution or
+        None if an invalid level was requested
+        """
+        return self._get_attribute(self.affine_invs, level)
+
+    def get_sigmas(self, level):
+        r"""Smoothing parameters used at a given level
+
+        Returns the smoothing parameters (a scalar for each axis) used at the
+        requested level of the scale space
+
+        Parameters
+        ----------
+        level : int, 0 <= from_level < L, (L = number of resolutions)
+            the scale space level to get the smoothing parameters from
+
+        Returns
+        -------
+        the smoothing parameters at the requested level
+
+        """
+        return self._get_attribute(self.sigmas, level)
+
+
+class IsotropicScaleSpace(ScaleSpace):
```

Comment:
And this is to build an ANTS2-like SS. This provides the user with more flexibility to control how the SS is built, although it may be harder to use...